internal/app/mapper/mapper.go

```diff
@@ -48,11 +50,19 @@ func New(config *aggregationv1.KeyerConfiguration, scope tally.Scope) Mapper {
 
 // GetKey converts a request into an aggregated key
 func (mapper *mapper) GetKey(request transport.Request) (string, error) {
-	if request.GetTypeURL() == "" {
-		mapper.scope.Counter(metrics.MapperError).Inc(1)
+	typeURL := request.GetTypeURL()
+	if typeURL == "" {
+		mapper.scope.Counter(metrics.MapperErrorEmptyURL).Inc(1)
 		return "", fmt.Errorf("typeURL is empty")
 	}
 
+	// A known issue (https://github.com/envoyproxy/envoy/issues/7529) causes envoy to generate
+	// EDS requests without resource_names, which could affect certain fragment rules.
+	if isEDS(typeURL) && len(request.GetResourceNames()) == 0 {
+		mapper.scope.Counter(metrics.MapperErrorEmptyResource).Inc(1)
```

Comment:
wdyt of adding a typeurl tag on the stat here https://github.com/envoyproxy/xds-relay/blob/master/internal/app/orchestrator/orchestrator.go#L141 ? We can alarm on all tags except eds.

Reply:
I believe the MapperErrorEmptyResource stat only increments for the EDS typeURL. On you second comment, if you mean adding a tag on this: `metrics.OrchestratorUnaggregatedWatchErrorsSubscope(o.scope).Counter(metrics.ErrorUnaggregatedKey).Inc(1)`, 
don't we then ignore eds errors caused by aggregation rule matching issues because we treat all error metrics from GetKey() as the same, but only differentiate by typeurl?

Reply:
Adding a tag will not change the aggregation, but yes it gives us the leverage to ignore it if necessary.

Reply:
What we have right now makes sense.I'm suggesting to add tags as an addition

Reply:
Per conversation offline, we decided to move forward with only maintaining the MapperError stat, ignoring the missing EDS resource names from stat counter.